IotcRestApi/regulator_draft.md

```diff
@@ -0,0 +1,145 @@
+
+# Chybové odpovědi
+Chyby 5xx jsou způsobené chybou serveru a neměli by nikdy nastat. Pokud server vratí tuto chybu tak
+by měl být okamžitě informován zástupce z firmy Netlia.
+
+Chyby 4xx jsou vráceny, pokud klient provedl neplatný/nevalidní request. Vysvětlení stavových kódu je možné najít
+v HTTP dokumentaci - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses. Podporované stavové kódy budou doplněny.
+
+V některých případech 4xx error obsahuje body s dalšimy informacemi o chybě. Body má následující formát:
+
+```yaml
+{
+    "errors" : [
+        {
+            "errorCode":int
+            "description":string
+            "additionalInformation":string
+            "traceId":string
+        }
+    ]
+}
+```
+
+# Zařízení a podporované endpointy
+
+## Teplotní regulátor
+
+### POST api/temperature-regulator/{DeviceId}/mode
+
+Změna módu teplotního regulátoru.
+
+Předávané parametry:
+
+| Parametr    | Typ         | Povinný | Popis                               |
+|:------------|:------------|:--------|:------------------------------------|
+| requestId   | string-guid | ano     | Jednoznačný identifikátor requestu. |
+| mode        | string      | ano     | Cílový mód zařízení.                |
+
+Cílový mód zařízení může nabývat těchto hodnot:
+
+| Hodnota            | Název                       |
+|--------------------|-----------------------------|
+| basic-regulator    | Základní regulace teploty.  |
+| manual             | Manuální regulace.          |
+| summer             | Letní režim.                |
+
+Ukázka requestu:
+
+```yaml
+{
+    "requestId": "b5e5a8e4-d09d-4d0f-8878-5ab24c2647fc",
+    "mode": "basic-regulator"
+}
+```
+
+### GET api/temperature-regulator/{DeviceId}/mode
+
+Zjištění módu teplotního regulátoru.
+
+Ukázka response:
+
+```yaml
+{
+    "mode": "basic-regulator"
+}
+```
+
+### POST api/temperature-regulator/{DeviceId}/temperature
+
+Nastavení cílové teploty pro regulaci.
+Podporováno pouze, pokud mód regulátoru je `basic-regulator`.
+
+Předávané parametry:
+
+| Parametr           | Typ         | Povinný | Popis                               |
+|:-------------------|:------------|:--------|:------------------------------------|
+| requestId          | string-guid | ano     | Jednoznačný identifikátor requestu. |
+| target-temperature | float       | ano     | Cílová teplota.                     |
+
+Ukázka requestu:
+
+```yaml
+{
+    "requestId": "b5e5a8e4-d09d-4d0f-8878-5ab24c2647fc",
+    "target-temperature": 21.5
+}
+```
+
+### GET api/temperature-regulator/{DeviceId}/temperature
+
+Zjištění cílové teploty pro regulaci.
+Podporováno pouze, pokud mód regulátoru je `basic-regulator`.
+
+Ukázka response:
```

Comment:
Pridal bych ukazku requestu.

Reply:
To je jen GET, co bychom tam ukazovali? Nebo udělat ke každému requestu kód pro CURL? To mi přijde zbytečné když budeme mít swagger.

Reply:
to mas asi pravdu. Technicky vzato GET muze mit i body ale vetsinu se to nepouziva. Tak ze je to zbytecne.